### DISCUSSION TITLE
How to install the latest version on Ubuntu and derivatives

### DISCUSSION THREAD

**Discussion by NoRainfallDuckDog** [netblue30/firejail#4663] [Q&A] [answered]
How do I get the current version of firejail (0.9.66) on my Mint system?

The version I have is 0.9.62:

* `firejail --version`: It shows firejail version 0.9.62
* `sudo apt install firejail`: It still shows 0.9.62

It didn't upgrade the version to 0.9.66.

Is there a way to get version 0.9.66 without upgrading from 0.9.62?

Would I delete or uninstall the one I have on Mint?

**Comment by kmk3** [comment, marked as answer]
@nntn-nntn [on Nov 2](https://github.com/netblue30/firejail/discussions/4663#discussion-3663504):

> How do I get the current version of firejail (0.9.66) on my Mint system?
> 
> The version I have is 0.9.62:
> 
> * `firejail --version`: It shows firejail version 0.9.62
> * `sudo apt install firejail`: It still shows 0.9.62

> 
> It didn't upgrade the version to 0.9.66
> 
> Is there a way to get version 0.9.66 without upgrading from 0.9.62?
> 
> Would I delete or uninstall the one I have on Mint?

From README.md:

> On Ubuntu 18.04 LTS users are advised to use the [PPA](https://launchpad.net/~deki/+archive/ubuntu/firejail).

Note: Consider that "Ubuntu 18.04 or later and derivatives".

From the PPA page:

>     sudo add-apt-repository ppa:deki/firejail
>     sudo apt-get update

(Continuing README.md)

> On Debian buster we recommend to use the [backports](https://packages.debian.org/buster-backports/firejail) package.
> 
> You can also install one of the [released
> packages](http://sourceforge.net/projects/firejail/files/firejail), or clone
> Firejail’s source code from our Git repository and compile manually:
> 
> `````
> $ git clone https://github.com/netblue30/firejail.git
> $ cd firejail
> $ ./configure && make && sudo make install-strip
> `````
> On Debian/Ubuntu you will need to install git and gcc compiler. AppArmor
> development libraries and pkg-config are required when using `--apparmor`
> ./configure option:
> `````
> $ sudo apt-get install git build-essential libapparmor-dev pkg-config gawk
> `````
> For `--selinux` option, add libselinux1-dev (libselinux-devel for Fedora).
> 
> Detailed information on using firejail from git is available on the [wiki](https://github.com/netblue30/firejail/wiki/Using-firejail-from-git).

Cc: @reinerh

---

Misc: I changed the discussion title:

> How to get version 9.66 without upgrading

->

> How to install the latest version on Ubuntu and derivatives

And category:

* General -> Q&A

**Reply by NoRainfallDuckDog** [reply]
I posted this before as follows:

https://github.com/netblue30/firejail/discussions/4500

So I deleted the appimage for Librewolf and used flatpak to retrieve it, but now I can't use firejail.

It has the same errors about the profile. So this is why I am asking.

It seems there is so much work that is required to get through this process.

What is really PPA? And why does it say "this untrusted PPA"?

I am new to this so pardon the amateur questions and the hesitance to follow through because I have other things to do, so I want a quick solution without the extra steps and without going through the README to get into the technical side of it. 

I can handle a few CLI commands.

I am using Mint and Ubuntu may be very different.

I will try to see what I can do or else I will have to go without firejail and use LW on its own. 

Thanks!

**Reply by kmk3** [reply]
@nntn-nntn [on Nov 4](https://github.com/netblue30/firejail/discussions/4663#discussioncomment-1590154):

> I posted this before as follows:
> 
> #4500
> 
> So I deleted the appimage for Librewolf and used flatpak to retrieve it, but
> now I can't use firejail.
> 
> It has the same errors about the profile. So this is why I am asking.
> 
> It seems there is so much work that is required to get through this process.

Installing firejail on Ubuntu/Mint should be as simple as this:

```sh
sudo add-apt-repository ppa:deki/firejail
sudo apt-get update
sudo apt-get install firejail firejail-profiles
```

Firejail is not compatible with snaps nor flatpaks.  But it should work with
appimages.  And considering that there is no official LibreWolf package for
Ubuntu, your best bet for using firejail with it is probably with the appimage.
It should work with something like this:

```sh
firejail --appimage --profile=librewolf ./LibreWolf.x86_64.AppImage
```

> What is really PPA?

A PPA can be thought of as a third-party software repository, commonly used to
provide packages for programs (or alternative builds of programs) that are not
available in the official (i.e.: distribution-provided) software repositories.
If you try to install a package normally (e.g.: with
`apt-get install firejail`), apt should look both in the official repository
sources as well as in your PPA sources (if any) and (if the package exists in
more than one source) choose the source with the higher priority (likely a
PPA).

> And why does it say "this untrusted PPA"?

Probably because a PPA can be made by anyone AFAIK (i.e.: it's not
official/it's not provided by the distribution).  Just like a .deb file can be
produced by anyone.  So only add PPAs from sources that you trust (like you
would only install software from sources that you trust).  See also the AUR
(for Arch Linux):

* <https://wiki.archlinux.org/title/Arch_User_Repository>

> I am new to this so pardon the amateur questions and the hesitant to follow
> through because I have other things to do, so I want a quick solution without
> the extra steps and without going through the README to get into the
> technical side of it.
> 
> I can handle a few CLI commands.
> 
> I am using Mint and Ubuntu may be very different.

I don't think it matters with regards to firejail.  In fact, Mint should
probably be easier because it does not use snaps by default (see #4554).

> I will try to see what I can do or else I will have to go without firejail
> and use LW on its own.
> 
> Thanks!

**Reply by kmk3** [reply]
Link to the discussion of installing from the PPA vs from the Ubuntu
repositories:

* <https://github.com/netblue30/firejail/discussions/4666>

Misc: This discussion and the above one were added to README.md: #4748
